### PR TITLE
[misc] Enable clang-tidy check in CI

### PR DESCRIPTION
**Pull request by sjwsl**
Related issue = #

### CODE REVIEW

File: .github/workflows/presubmit.yml [hunk 1]
```diff
@@ -98,6 +98,36 @@ jobs:
...
+        env:
+          CR_PAT: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run clang-tidy
+        run: |
+          if [[ ${{needs.check_files.outputs.run_job}} == false ]]; then
+            exit 0
+          fi
+          docker run -id --user dev --name check_clang_tidy ghcr.io/taichi-dev/taichidev-cpu-ubuntu18.04:v0.1.0 /bin/bash
+          tar -cf - ../${{ github.event.repository.name }} --mode u=+rwx,g=+rwx,o=+rwx --owner dev --group dev | docker cp - check_clang_tidy:/home/dev/
+          docker exec --user root check_clang_tidy apt install -y clang-tidy-10
```
Comment (ailzhang) [inline .github/workflows/presubmit.yml:126]: Can we add a todo here saying that we should install clang-tidy-10 inside the docker image when we rebuild ;) ?

**Review by ailzhang — APPROVED** [review]
Nice work! Btw I took a look at our log and it's a bit too verbose :P . Can we have a followup that: 
- If clang-tidy passes, let's just print one sentence and celebrate
- If it fails, clearly list the offending file names and a command to reproduce.

File: .github/workflows/presubmit.yml (hunk 1 above)
Comment (sjwsl) [reply .github/workflows/presubmit.yml:127]: Finished! Thanks :)

**Review by ailzhang — APPROVED** [review]
Awesome thanks!

**Pull request merged**